solr/core/src/java/org/apache/solr/schema/BloomStrField.java

```diff
@@ -0,0 +1,206 @@
+  }
+
+  @Override
+  public boolean isPolyField() {
+    return true;
+  }
+
+  /**
+   * Base suffix must start with `0` to ensure that FieldsProducer visits the subfield immediately
+   * after the parent field -- i.e., the fieldnames must sort lexicographically adjacent.
+   */
+  public static final String BLOOM_FIELD_BASE_SUFFIX = "0NgramS"; // single-valued
+
+  public static final String BLOOM_FIELD_BASE_SUFFIX_MULTI = "0NgramM"; // multiValued
+
+  private IndexSchema schema;
+  private FieldType bloomFieldType;
+  private static final Analyzer KEYWORD_ANALYZER =
+      new Analyzer() {
+        @Override
+        protected TokenStreamComponents createComponents(String fieldName) {
+          Tokenizer tk = new KeywordTokenizer();
+          return new TokenStreamComponents(tk, tk);
+        }
+      };
+
+  @Override
+  protected void init(IndexSchema schema, Map<String, String> args) {
+    this.schema = schema;
+    String bloomAnalyzerId = args.remove("bloomAnalyzerId");
+    String postingsFormat = args.get("postingsFormat");
+    PostingsFormat pf;
+    if (postingsFormat != null) {
+      pf = PostingsFormat.forName(postingsFormat);
+    } else {
+      // start with the default postingsFormat.
+      pf = Codec.getDefault().postingsFormat();
+      if (pf instanceof PerFieldPostingsFormat) {
+        pf = ((PerFieldPostingsFormat) pf).getPostingsFormatForField("");
+      }
+    }
+    if (pf instanceof BloomAnalyzerSupplier) {
+      if (bloomAnalyzerId != null
+          && !bloomAnalyzerId.equals(((BloomAnalyzerSupplier) pf).getBloomAnalyzerId())) {
+        throw new IllegalArgumentException(
+            "specified `bloomAnalyzerId` "
+                + bloomAnalyzerId
+                + " conflicts with the `bloomAnalyzerId` of the specified `postingsFormat`");
+      }
+    } else {
+      if (bloomAnalyzerId == null) {
+        bloomAnalyzerId = DEFAULT_BLOOM_ANALYZER_ID;
+      }
+      pf = PostingsFormat.forName(constructName(pf, bloomAnalyzerId));
+      if (!(pf instanceof BloomAnalyzerSupplier)) {
+        throw new IllegalArgumentException(
+            "constructed `postingsFormat` does not support ngram bloom filter: " + pf);
+      }
+      // replace any existing postingsFormat spec with our the constructed postingsFormat
+      args.put("postingsFormat", pf.getName());
+    }
+    bloomFieldType = getFieldType(schema, (PostingsFormat & BloomAnalyzerSupplier) pf);
+    super.init(schema, args);
+  }
+
+  private <T extends PostingsFormat & BloomAnalyzerSupplier> FieldType getFieldType(
+      IndexSchema schema, T pf) {
+    Map<String, String> props = new HashMap<>();
+    props.put("indexed", "true");
+    props.put("stored", "false");
+    props.put("docValues", "false");
+    props.put("sortMissingLast", "true");
+    props.put("termVectors", "false");
+    props.put("omitNorms", "true");
+    props.put("omitTermFreqAndPositions", "false");
+    props.put("uninvertible", "true");
```

Comment:
@magibney `props.put("uninvertible", "true");` , was just looking this, here we want `false` ?

Reply:
whoa! good catch, thank you! will fix this.